drivers/lcd/lcd.c

```diff
@@ -25,45 +27,163 @@
 #include "byteorder.h"
 #include "periph/spi.h"
 #include "kernel_defines.h"
+#include "ztimer.h"
 
 #include "lcd.h"
 #include "lcd_internal.h"
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-static void _lcd_spi_acquire(const lcd_t *dev)
+static inline void _lcd_write_byte(const lcd_t *dev, bool cont, uint8_t data)
 {
-    spi_acquire(dev->params->spi, dev->params->cs_pin, dev->params->spi_mode,
-                dev->params->spi_clk);
+    if (dev->params->spi != SPI_UNDEF) {
+        /* SPI serial interface is used */
+        spi_transfer_byte(dev->params->spi, dev->params->cs_pin, cont, data);
+    }
+    else {
+        assert(false);
```

Comment:
Why not just use `assert(dev->params->spi != SPI_UNDEF);` at the beginning of the function? (Here and below)

Reply:
Yeah, the reason is that this change is a split-off from parallel interface implementation where it will be
```c
#if IS_USED(MODULE_LCD_SPI)
    if (dev->params->spi != SPI_UNDEF) {
        /* SPI serial interface is used */
        spi_transfer_byte(dev->params->spi, dev->params->cs_pin, cont, data);
    }
    else {
#endif
#if IS_USED(MODULE_LCD_PARALLEL)
        /* MCU 8080 8-/16-bit parallel interface is used */
        _lcd_write_byte_par(dev, cont, data);
#else
        assert(false);
#endif
#if IS_USED(MODULE_LCD_SPI)
    }
#endif
```

I can change it, if you want :smile:

Reply:
Let's keep it like this if it's easier for you to rebase once that's merged.